modules/agar/src/test/ts/browser/keyboard/TypeInInputTest.ts

```diff
@@ -0,0 +1,140 @@
+import { Assert, describe, it } from '@ephox/bedrock-client';
+import { DomEvent, Insert, Remove, SugarBody, SugarElement } from '@ephox/sugar';
+
+import * as TestStore from 'ephox/agar/api/TestStore';
+import * as UiControls from 'ephox/agar/api/UiControls';
+
+interface InputStoreDataItem {
+  readonly type: 'beforeinput' | 'input';
+  readonly inputType: string;
+  readonly data: string;
+}
+
+interface KeyboardStoreDataItem {
+  readonly type: 'keydown' | 'keypress' | 'keyup';
+  readonly which: number;
+  readonly key: string;
+  readonly code: string;
+  readonly keyCode: number;
+  readonly charCode: number;
+  readonly shiftKey: boolean;
+}
+
+type StoreDataItem = InputStoreDataItem | KeyboardStoreDataItem;
+
+describe('browser.agar.keyboard.TypeInInputTest', () => {
+  const pTestOnTag = async (tagHtml: string) => {
+    const store = TestStore.TestStore<StoreDataItem>();
+
+    const input = SugarElement.fromHtml<HTMLInputElement | HTMLTextAreaElement>(tagHtml);
+    Insert.append(SugarBody.body(), input);
+
+    DomEvent.bind(input, 'keydown', ({ raw, kill }) => {
+      if (raw.keyCode === 68) {
```

Comment:
I don't think we should use `keyCode` in new code since it's deprecated and less readable. I think it's better that we use the `code` property. What do you think ?

Reply:
Yes, that makes sense since at some point we might want to add some lint rule to get rid of these and keeping it around would just add work to that. I will see if I can stick in a follow up ticket to also use this more modern key handling for the `Keyboard.keydown` etc should be pretty simple to introduce it there then we can simulate modern keyboard events with agar that would be nice.